scripts/lib/CIME/case_st_archive.py

```diff
@@ -21,15 +21,25 @@ def _get_archive_file_fn(copy_only):
     return shutil.copyfile if copy_only else shutil.move
 
 ###############################################################################
-def _get_datenames(rundir, casename):
+def _get_datenames(rundir, casename, case):
```

Comment:
If you are passing in the case, why also pass in rundir and casename which have just been extracted from the case by the caller? I suggest just passing in case and move the case.get_value calls into `_get_datenames`.